prepare_data/src/main/groovy/org/orbisgis/osm/FormattingForAbstractModel.groovy

```diff
@@ -0,0 +1,369 @@
+    String b_r_lev = row.getString("b_r_lev")
+    int result = 0
+    if (b_lev != null && b_lev.isFloat()) {
+        if ((r_lev != null && r_lev.isFloat()) || (b_r_lev != null && b_r_lev.isFloat())) {
+            if (r_lev.isFloat()) {
+                result = b_lev.toFloat() + r_lev.toFloat()
+            } else {
+                result = b_lev.toFloat() + b_r_lev.toFloat()
+            }
+        } else {
+            result = b_lev.toFloat()
+        }
+    }
+    return result
+}
+
+/**
+ * This function defines the value of the column width according to the value of width from OSM
+ * @param width The original width value
+ * @return the calculated value of width (default value : null)
+ */
+static Float getWidth (String width){
+    Float result
+    if (width != null && width.isFloat()) {
+        result=width.toFloat()
+    }
+    return result
+}
+
+/**
+ * This function defines the value of the column zindex according to the value of zindex from OSM
+ * @param width The original zindex value
+ * @return The calculated value of zindex (default value : null)
+ */
+static Integer getZIndex (String zindex){
+    Integer result
+    if (zindex != null && zindex.isInteger()) {
+        result=zindex.toInteger()
+    }
+    return result
+}
+
+/**
+ * This function defines the input value for a column of the geoClimate Input Model according to a given mapping between the expected value and the set of key/values tag in OSM
+ * @param row The row of the raw table to examine
+ * @param myMap A map between the target values in the model and the associated key/value tags retrieved from OSM
+ * @return A list of Strings : first value is for "type" and second for "use"
+ */
+static String getAbstractValue(def row, def myMap) {
+    String strType = null
+    myMap.each { finalVal ->
+        finalVal.value.each { osmVals ->
+            osmVals.value.each { osmVal ->
+                def val = osmVal.toString()
+                if (val.startsWith("!")) {
+                    val.replace("! ","")
+                    if ((row.getString(osmVals.key) != val) && (row.getString(osmVals.key) != null)) {
+                        if (strType == null) {
+                            strType = finalVal.key
+                        }
+                    }
+                } else {
+                    if (row.getString(osmVals.key) == val) {
+                        if (strType == null) {
+                            strType = finalVal.key
+                        }
+                    }
+                }
+            }
+        }
+    }
+    return strType
+}
+
+/**
+ * This function defines the value of the column sidewalk according to the values of sidewalk from OSM
+ * @param width The original sidewalk value
+ * @return The calculated value of sidewalk (default value : null)
+ */
+static String getSidewalk(String sidewalk) {
+    String result
+    if (sidewalk != null) {
+        if (sidewalk == 'both') {
```

Comment:
The multiple if can be replaced by a switch like in this case :
``` groovy
switch(sidewalk){
    case 'both':
        result = 'two'
        break
    case 'right':
    case 'left':
    case 'yes':
        result = 'one'
        break
    default:
        result = 'no'
        break
}
```

Reply:
Done